docs/README.md

```diff
@@ -0,0 +1,624 @@
+
+In short, the authentication flow looks like the following:
+1. Exchange a Refresh Token for an ID Token with Firebase
+2. Use the ID Token with the AIM API
+3. Repeat step 1 as the previous ID Token expires
+
+![Authentication Flow](./authentication_flow.png)
+
+Once acquired, the ID Token must be sent in the `Authorization` HTTP Header as a `Bearer` token.
+
+As ID Tokens are short lived, a new one must be fetched before expiration and replaced in requests to the API. ID Tokens are [JSON Web Tokens](https://jwt.io/), so any standard JWT libary can be used to decode them and inspect the `exp` entry for a Unix timestamp after which the token will be rejected by the API. A number of JWT libraries are referenced in the link above.
+
+### Obtain an ID Token
+
+In order to obtain an ID Token, use the `id_token` url from the [Discovery document](#api-discovery), which allows us to exchange our Refresh Token for a fresh ID Token.
+
+<a class="id_token-url"></a>
+
+Make a POST request with the following payload, injecting the Refresh Token as specified: `{"grant_type": "refresh_token", "refresh_token": <API Key>}`. Extract the `id_token` field from the response, which contains the ID Token, which can then be sent to the API.
+
+For example, with `curl` to make the request and `jq` to extract the field:
+
+```sh
+ID_TOKEN_URL="<ID Token Discovery URL>"
+API_KEY="<Refresh Token>"
+curl -fsSl -XPOST \
+     -H "Content-Type: application/json" \
+     -d "{\"grant_type\": \"refresh_token\", \"refresh_token\": \"$API_KEY\"}" \
+     "$ID_TOKEN_URL" \
+     | jq -r '.id_token'
+```
+
+You now have an ID Token that can be used with the AIM API!
+
+### Abuse and Privacy
+
+In order to prevent abuse and data leaks, Refresh Tokens must be stored securely. In particular, avoid unneeded sharing of Refresh Tokens or storing them in source files/source control.
+
+If you suspect your Refresh Token has been compromised, contact <mailto:AIM@thestrawgroup.com> as soon as possible. If abuse is suspected, TSG may disable Refresh Tokens immediately and follow up with the API contact.
+
+## Query API
+
+The Query API is the primary tool provided by the AIM API. It provides powerful data analysis across multiple dimensions of the AIM dataset.
+
+The Query API is comprised of four (4) primary components:
+- `aggregation`
+- `attribute`
+- `metric`
+- `normalization`
+
+Each component has a discovery endpoint to obtain the available items with full metadata. All query urls are based from the `warehouse` url in the [discovery document](#api-discovery):
+
+<code class="warehouse-url"></code>
+
+### Quickstart
+
+After [acquiring an ID Token](#obtain-an-id-token), start with a few simple API calls. The calls will use `curl` for demonstration, but of course, any HTTP client will do. In these examples, `BASE_URL` is set to <code class="warehouse-url"></code>. Any query results are for demonstration purposes only and do not represent real values.
```

Comment:
Al:
> On curl mentioned above should it say “curl commands”?

Reply:
I think this is another thing that looks better on the website (than the copy in Word) - `curl` is wrapped as code. We can make this change if you still think so.